### PR TITLE
Set up the KtBridge Kotlin compiler plugin builds

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,9 +14,15 @@ buildscript {
   }
 }
 
+plugins {
+  id 'com.gradle.plugin-publish' version "0.12.0" apply false
```
Comment (JakeWharton) [inline build.gradle]:
```suggestion
```

File: ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradlePlugin.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    groupId = BuildConfig.KOTLIN_PLUGIN_GROUP,
+    artifactId = BuildConfig.KOTLIN_PLUGIN_NAME,
+    version = BuildConfig.KOTLIN_PLUGIN_VERSION
+  )
+
+  override fun getPluginArtifactForNative(): SubpluginArtifact = SubpluginArtifact(
+    groupId = BuildConfig.KOTLIN_PLUGIN_GROUP,
+    artifactId = BuildConfig.KOTLIN_PLUGIN_NAME + "-native",
+    version = BuildConfig.KOTLIN_PLUGIN_VERSION
+  )
```
Comment (JakeWharton) [inline ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradlePlugin.kt]:
```suggestion
```

The build for this artifact is not set up

File: ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradleExtension.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package app.cash.quickjs.ktbridge.plugin
+
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+
+open class KtBridgeGradleExtension(objects: ObjectFactory) {
```
Comment (JakeWharton) [inline ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradleExtension.kt:23]: I don't see any usage of the extension. Delete it all until needed?

File: ktbridge/plugin/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+plugins {
+  kotlin("jvm")
+  kotlin("kapt")
+  id("com.github.gmazzo.buildconfig")
+}
+
+dependencies {
+  compileOnly("org.jetbrains.kotlin:kotlin-compiler-embeddable")
```
Comment (JakeWharton) [inline ktbridge/plugin/build.gradle.kts:8]: This module should also have a `gradle.properties` with:
```properties
# We get the stdlib transitively as a compileOnly dependency.
kotlin.stdlib.default.dependency=false
```

File: ktbridge/plugin/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeCommandLineProcessor.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.google.auto.service.AutoService
+import org.jetbrains.kotlin.compiler.plugin.AbstractCliOption
+import org.jetbrains.kotlin.compiler.plugin.CliOption
+import org.jetbrains.kotlin.compiler.plugin.CommandLineProcessor
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.config.CompilerConfigurationKey
+
+@AutoService(CommandLineProcessor::class)
+class KtBridgeCommandLineProcessor : CommandLineProcessor {
```
Comment (JakeWharton) [inline ktbridge/plugin/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeCommandLineProcessor.kt:27]: Can also nuke this whole type

**Review by JakeWharton — APPROVED** [review]
Quite a lot of things we don't need in here

**Comment by swankjesse** [comment]
I’m gonna make a big follow-up PR instead of a chain of rebases.

File: ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradleExtension.kt (hunk 3 above)
Comment (swankjesse) [reply ktbridge/plugin-gradle/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeGradleExtension.kt:23]: Deleted!

File: ktbridge/plugin/build.gradle.kts (hunk 4 above)
Comment (swankjesse) [reply ktbridge/plugin/build.gradle.kts:8]: Ooooh I like that.

File: ktbridge/plugin/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeCommandLineProcessor.kt (hunk 5 above)
Comment (swankjesse) [reply ktbridge/plugin/src/main/kotlin/app/cash/quickjs/ktbridge/plugin/KtBridgeCommandLineProcessor.kt:27]: Nuked.

**Pull request merged**